### ISSUE TITLE
Error RPI, 4Go, Boot USB impossible

### ISSUE THREAD

**Issue by gado070707** [maxnet/berryboot#684]
Hello,
I just installed berryboot on a raspberrypi4 by booting from the USB.
When launching the Raspbian OS, I got an error message "Timed out waiting for device / dev / disk / by-label / boot.".

Step : A start job is running for /dev/disk/by-label/boot (26s / 1min 30s)
then
[ TIME ] Timed out waiting for device /dev/disk/by-label/boot.
[DEPEND] Dependency failed for /boot
[DEPEND] Dependency failed for Local File systems.

Do you have a solution ?
Thank you in advance.

![error](https://user-images.githubusercontent.com/14954799/105138281-f5f61f80-5af4-11eb-9a19-1d498398eb37.png)
![error2](https://user-images.githubusercontent.com/14954799/105138284-f7274c80-5af4-11eb-87d2-57b8d9c42e96.png)

**Comment by andreaz70** [comment]
I am running into the same, after installing on a iscsi disk.

**Comment by spammassive** [comment]
![IMG_20210125_144134](https://user-images.githubusercontent.com/77996499/105770906-0b1be580-5f1d-11eb-9fef-59503e292eec.jpg)
Same here: RPi (full 64-bit) won't boot (see attached pic).
Ubuntu, Manjaro and OpenSuse work well though.

Edit: Downloading official RasPiOs from raspberrypi.org and installing it has worked for me.

**Comment by NeonFux** [comment]
I have same problem with Debian 64 and New 8gb Rasp

**Comment by drtrotter74** [comment]
Same for me.....Manjaro works, but I have wifi issues, and get some strange messages also when the bootloader comes up.  Has anyone tried going back to a previous version of BerryBoot to see if it works better?

**Comment by drtrotter74** [comment]
My other question is why does BerryBoot 2.0 only have a few OS's available.  On some other youtube vlogs, I saw BB with a bunch of different OS's available some 32 bit

**Comment by neoavalon** [comment]
Same boot issue w/ Buster + 2GB Pi4B + encrypted USB drive

**Comment by owl770** [comment]
This sounds like an issue with the image itself (rather than Berryboot). You would need to unsquash the image, edit `/etc/fstab` (to comment out the line that starts with `LABEL=boot`), resquash the image and then try booting.

**Comment by spammassive** [comment]
> This sounds like an issue with the image itself (rather than Berryboot). You would need to unsquash the image, edit `/etc/fstab` (to comment out the line that starts with `LABEL=boot`), resquash the image and then try booting.

I had this issue with the RPiOS which I got through berryboot. If it was a bad image, the issue should happen to everyone who got RPiOS via berryboot, which would be a lot of berryboot users, don't you think?

**Comment by tristan-k** [comment]
Same issue here.

**Comment by adamflorizone** [comment]
Same issue pi3

**Comment by maasnet** [comment]
Same issue with pi4 and ssd

**Comment by dtay4413** [comment]
I'm having same issue ubuntu installs and works fine but can't use raspberry pi os same error messages

**Comment by dev-mansonthomas** [comment]
Same issue here : https://github.com/maxnet/berryboot/issues/707


@spammassive  
> Edit: Downloading official RasPiOs from raspberrypi.org and installing it has worked for me.

How did you install it ? On an USB drive as well ?

**Comment by spammassive** [comment]
yes, via USB disk

**Comment by dev-mansonthomas** [comment]
@owl770 
> This sounds like an issue with the image itself (rather than Berryboot). You would need to unsquash the image, edit `/etc/fstab` (to comment out the line that starts with `LABEL=boot`), resquash the image and then try booting.

for raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot.img.gz
the content of the /etc/fstab is 

```
proc            /proc           proc    defaults          0       0
#PARTUUID=e8af6eb2-01  /boot           vfat    defaults          0       2
#PARTUUID=e8af6eb2-02  /               ext4    defaults,noatime  0       1
```

```bash
brew install squashfs
gunzip raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot.img.gz
sudo unsquashfs -f -d raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot  raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot.img
sudo cat raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot/etc/fstab
```


@spammassive  any speed walkthrough on how you did install on your usb device ? (I'm new to Raspberry Pi, so it may be obvious, but not for me :)

**Comment by spammassive** [comment]
just follow the instructions on berryboot website on how to add/install your own OS

**Comment by dev-mansonthomas** [comment]
@spammassive  are you referring to this ? 

https://www.berryterminal.com/doku.php/berryboot/adding_custom_distributions

**Comment by spammassive** [comment]
yes, that's it (if I remember correctly)

**Comment by owl770** [comment]
> @owl770
> 
> > This sounds like an issue with the image itself (rather than Berryboot). You would need to unsquash the image, edit `/etc/fstab` (to comment out the line that starts with `LABEL=boot`), resquash the image and then try booting.
> 
> for raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot.img.gz
> the content of the /etc/fstab is
> 
> ```
> proc            /proc           proc    defaults          0       0
> #PARTUUID=e8af6eb2-01  /boot           vfat    defaults          0       2
> #PARTUUID=e8af6eb2-02  /               ext4    defaults,noatime  0       1
> ```
> 
> ```shell
> brew install squashfs
> gunzip raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot.img.gz
> sudo unsquashfs -f -d raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot  raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot.img
> sudo cat raspberry_pi_os_buster_lite_all_rpi_2021.01.11_berryboot/etc/fstab
> ```
> 
> @spammassive any speed walkthrough on how you did install on your usb device ? (I'm new to Raspberry Pi, so it may be obvious, but not for me :)

It might be worth checking what is in /boot/cmdline.txt and comparing that to the disk labels on your boot and root partitions.

**Comment by dev-mansonthomas** [comment]
Thanks to @spammassive I've built a "custom image" for berryboot.
I've described here how to do that : 
https://github.com/maxnet/berryboot/issues/707#issuecomment-850844202

There's a link to a full documented Raspberry Pi Cluster I'm building (work in progress)

There's a link to download the custom image I've built (it's only the lite version, you'll have to do the same with the desktop version (or ask me nicely))

**Comment by jejones3141** [comment]
Same issue here installing RPIOS 64-bit from wherever berryboot gets it by default, on an RPi. Could a working image be put there so it only has to be redone once?

**Comment by maxnet** [comment]
Should work better in 2021-07-01

**Comment by AleXSR700** [comment]
I get the same behavior from stock Raspberry OS downloaded from Raspberry org.
Downloaded newest Pi Imager and latest 64 bit Raspberry OS ( end of May) and am having the same issue after reboot. Performed update and upgrade commands. 
System worked at first and then starts failing on check. I think it was the fifth reboot or so.

fstab and cmdline look fine.  IDs are correct. 

I first installed EEPROM via Pi Imager with an SD Card and selected USB boot. Then the Os to my HDD (not SSD).

Any information on how to fix this?

**Comment by thomai-d** [comment]
After booting a freshly installed raspi 4B from SSD only, I got the same error: `Dependency failed for /boot`.

**Using an SATA-USB adapter with external power fixed the issue for me.**

Not sure if it's because I use a non-official power supply with 5V / 3A (the original one has 5.1 / 3A spec).

But what's strange is, that the same hardware worked perfectly before I reinstalled the system ... ¯\_(ツ)_/¯

**Comment by adamflorizone** [comment]
Keep in mind this is issue has been closed, and since its opening the official RPi boot loader was updated to support USB, SD and net boot. See: https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#raspberry-pi-4-bootloader-configuration . 

I ended up switching to the updated boot loader (abandoning berryboot) and had no issues with ssd or netboot. It has been recommended to use a powered hub since not all drives can be powered up by the RPi 4. ONE of my HDDs did boot and ran for years on the USB 3 port.
